sprint-mission-6.sql

```diff
@@ -0,0 +1,220 @@
+select * 
+from orders 
+order by date desc, time desc;
+
+-- 5. `orders` 테이블에서 오프셋 기반 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일 때, 첫 번째 페이지를 조회하세요.
+select * 
+from orders 
+order by date desc, time desc 
+limit 10;
+
+-- 6. `orders` 테이블에서 오프셋 기반 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일 때 5번째 페이지를 조회하세요.
+select * 
+from orders 
+order by date desc, time desc 
+limit 10 offset 40;
+
+-- 7. `orders` 테이블에서 커서 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일때, `id` 값을 기준으로 커서를 사용합시다. 커서의 값이 `42`일 때 다음 페이지를 조회하세요.
+select * 
+from orders 
+where id < 42 
+order by id desc 
+limit 10;
+
+-- 8. `orders` 테이블에서 2025년 3월에 주문된 내역만 조회하세요.
+select * 
+from orders 
+where date >= '2025-03-01' and date < '2025-04-01';
+
+-- 9. `orders` 테이블에서 2025년 3월 12일 오전에 주문된 내역만 조회하세요.
+select * 
+from orders 
+where date = '2025-03-12' and time < '12:00:00';
+
+-- 10. `pizza_types` 테이블에서 이름에 'Cheese' 혹은 'Chicken'이 포함된 피자 종류를 조회하세요. (대소문자를 구분합니다)
+select * 
+from pizza_types 
+where name like '%Cheese%' or name like '%Chicken%';
+
+-- # 중급 문제
+
+-- 1. `order_details` 테이블에서 각 피자(`pizza_id`)별로 주문된 건 수(`order_id`)를 보여주세요.
+select pizza_id, count(order_id) as "주문 건 수" 
+from order_details 
+group by pizza_id;
+
+-- 2. `order_details` 테이블에서 각 피자(`pizza_id`)별로 총 주문 수량을 구하세요.
+select pizza_id, sum(quantity) as "총 주문 수" 
+from order_details 
+group by pizza_id;
+
+-- 3. `pizzas` 테이블에서 `price`의 크기가 20보다 큰 피자의 종류만 `order_details` 테이블에서 조회하세요. (힌트: 서브쿼리)
+select pizza_id 
+from order_details 
+where pizza_id in (select id 
+    from pizzas 
+    where price > 20);
+
+-- 4. `orders` 테이블에서 각 날짜별 총 주문 건수를 `order_count` 라는 이름으로 계산하고, 하루 총 주문 건수가 80건 이상인 날짜만 조회한 뒤, 주문 건수가 많은 순서대로 정렬하세요.
+select date, count(id) as order_count 
+from orders 
+group by date 
+having count(id) > 80 
+order by order_count desc;
+
+-- 5. `order_details` 테이블에서 피자별(`pizza_id`) 총 주문 수량이 10개 이상인 피자만 조회하고, 총 주문 수량 기준으로 내림차순 정렬하세요.
+select pizza_id, sum(quantity) as "총 주문 수량" 
+from order_details 
+group by pizza_id 
+having count(quantity) > 10 
+order by "총 주문 수량" desc; 
+
+-- 6. `order_details` 테이블에서 피자별 총 수익을 `total_revenue` 라는 이름으로 구하세요. (수익 = `quantity * price`)
+select o.pizza_id, sum(o.quantity * p.price) as "total_revenue" 
+from order_details as o 
+join pizzas as p 
+on o.pizza_id = p.id 
+group by o.pizza_id; 
+
+-- 7. 날짜별로 피자 주문 건수(`order_count`)와 총 주문 수량(`total_quantity`)을 구하세요.
+select o.date, count(distinct od.order_id) as "order_count", sum(od.quantity) as "total_quantity"
+from order_details as od
+join orders as o
+on od.order_id = o.id
+group by o.date
+order by o.date desc;
+
+-- # 고급 문제
+
+
+--    1. 피자별(`pizzas.id` 기준) 판매 수량 순위에서 피자별 판매 수량 상위에 드는 베스트 피자를 10개를 조회해 주세요. `pizzas`의 모든 컬럼을 조회하면서 각 피자에 해당하는 판매량을 `total_quantity`라는 이름으로 함께 조회합니다.
+        select p.*, sum(od.quantity) as "total_quantity"
+        from pizzas as p
+        join order_details as od
+        on od.pizza_id = p.id
+        group by p.id
+        order by "total_quantity" desc
+        limit 10;
```

Comment:
풀스캔을 피하고, PSQL은 SELECT의 비집계 컬럼을 모두 GROUP BY에 포함해야해요

```suggestion
SELECT p.id, p.type_id, p.size, p.price,
       SUM(od.quantity) AS total_quantity
FROM pizzas p
JOIN order_details od ON p.id = od.pizza_id
GROUP BY p.id, p.type_id, p.size, p.price
ORDER BY total_quantity DESC
LIMIT 10;

```